sequencer/src/bin/submit-transactions.rs

```diff
@@ -0,0 +1,161 @@
+struct ParseSizeError {
+    msg: String,
+}
+
+fn parse_size(s: &str) -> Result<usize, ParseSizeError> {
+    Ok(s.parse::<ByteSize>()?.0 as usize)
+}
+
+#[async_std::main]
+async fn main() {
+    setup_backtrace();
+    setup_logging();
+
+    let opt = Options::parse();
+    let (sender, mut receiver) = mpsc::channel(opt.channel_bound);
+
+    // Subscribe to block stream so we can check that our transactions are getting sequenced.
+    let client = Client::<Error>::new(opt.url.clone());
+    let block_height: usize = client
+        .get("status/latest_block_height")
+        .send()
+        .await
+        .unwrap();
+    let mut blocks = client
+        .socket(&format!("availability/stream/blocks/{block_height}"))
+        .subscribe()
+        .await
+        .unwrap();
+    tracing::info!("listening for blocks starting at {block_height}");
+
+    // Spawn tasks to submit transactions.
+    for _ in 0..opt.jobs {
+        spawn(submit_transactions(opt.clone(), sender.clone()));
+    }
+
+    // Keep track of the results.
+    let mut pending = HashSet::new();
+    while let Some(block) = blocks.next().await {
+        let block: BlockQueryData<SeqTypes> = match block {
+            Ok(block) => block,
+            Err(err) => {
+                tracing::warn!("error getting block: {err}");
+                continue;
+            }
+        };
+        tracing::info!("got block {}", block.height());
+
+        // Get all transactions which were submitted before this block.
+        while let Ok(Some(tx)) = receiver.try_next() {
+            pending.insert(tx);
+        }
+
+        // Clear pending transactions from the block.
+        for (_, tx) in block.payload().enumerate() {
+            if pending.remove(&tx.commit()) {
+                tracing::debug!("got transaction {}", tx.commit());
+            }
+        }
+
+        tracing::info!("{} transactions still pending", pending.len());
+    }
+    tracing::info!(
+        "block stream ended with {} transactions still pending",
+        pending.len()
+    );
+}
+
+async fn submit_transactions(opt: Options, mut sender: Sender<Commitment<Transaction>>) {
+    let client = Client::<Error>::new(opt.url);
+    loop {
+        let tx = random_transaction(opt.min_size, opt.max_size);
+        let hash = tx.commit();
+        tracing::debug!(
+            "submitting transaction {hash} for namespace {} of size {}",
+            tx.vm(),
+            tx.payload().len()
+        );
+        if let Err(err) = client
+            .post::<()>("submit/submit")
+            .body_binary(&tx)
+            .unwrap()
+            .send()
+            .await
+        {
+            tracing::error!("failed to submit transaction: {err}");
+        }
+        sender.send(hash).await.ok();
+
+        if let Some(delay) = opt.delay {
+            sleep(delay).await;
+        }
+    }
+}
+
+fn random_transaction(min_size: usize, max_size: usize) -> Transaction {
+    let mut rng = thread_rng();
```

Comment:
How does this work with multithreading? It might be nice to allow passing in a seed to make the generated transactions deterministic, but we can also add that later.